src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollectorHostedService.cs

```diff
@@ -49,20 +55,24 @@ private async Task ExecuteAsync(CancellationToken cancellationToken)
 
             while (!cancellationToken.IsCancellationRequested)
             {
-                _logger.LogDebug("Executing HealthCheck collector HostedService.");
-
-                using (var scope = scopeFactory.CreateScope())
+                // Collect should not be triggered until IServerAddressFeature reports the listening endpoints
+                if (_serverAddressesService.HasAddresses)
                 {
-                    try
-                    {
-                        var runner = scope.ServiceProvider.GetRequiredService<IHealthCheckReportCollector>();
-                        await runner.Collect(cancellationToken);
+                    _logger.LogDebug("Executing HealthCheck collector HostedService.");
 
-                        _logger.LogDebug("HealthCheck collector HostedService executed successfully.");
-                    }
-                    catch (Exception ex)
+                    using (var scope = scopeFactory.CreateScope())
                     {
-                        _logger.LogError(ex, $"HealthCheck collector HostedService throw a error: {ex.Message}");
+                        try
+                        {
+                            var runner = scope.ServiceProvider.GetRequiredService<IHealthCheckReportCollector>();
+                            await runner.Collect(cancellationToken);
+
+                            _logger.LogDebug("HealthCheck collector HostedService executed successfully.");
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogError(ex, $"HealthCheck collector HostedService throw a error: {ex.Message}");

```

Comment:
BTW strings used with logging shouldn't use `$""`. Logging has its own format for capturing values.

```C#
_logger.LogError(ex, $"HealthCheck collector HostedService throw a error: {ExceptionMessage}, ex.Message, ex);
```